linker/shared/src/main/scala/org/scalajs/linker/frontend/Desugarer.scala

```diff
@@ -138,6 +139,20 @@ private[linker] object Desugarer {
         case prop: LinkTimeProperty =>
           coreSpec.linkTimeProperties.transformLinkTimeProperty(prop)
 
+        case LinkTimeIf(cond, thenp, elsep) =>
+          val cond1 = transform(cond)
+          coreSpec.linkTimeProperties.tryEvalLinkTimeBooleanExpr(cond1) match {
+            case Some(result) =>
+              if (result)
+                transform(thenp)
+              else
+                transform(elsep)
+            case None =>
+              throw new AssertionError(
+                  s"Invalid link-time condition should not have passed the reachability analysis: " +
+                  tree.show)
```

Comment:
Paired comment with the one in Infos: IIUC we can get here if ir checking is disabled and there is bad IR. IMO this is totally OK, but we might want to add some guidance in the error message to enable IR checking before reporting a bug.